specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/stable/2024-06-30/cloudhsm.json

```diff
@@ -0,0 +1,1441 @@
+            }
+          }
+        },
+        "x-ms-examples": {
+          "CloudHsmCluster_Create_Backup_MaximumSet_Gen": {
+            "$ref": "./examples/CloudHsmCluster_CreateOrValidate_Backup_MaximumSet_Gen.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HardwareSecurityModules/cloudHsmClusters/{cloudHsmClusterName}/backup/{jobId}/pending": {
+      "get": {
+        "tags": [
+          "Cloud HSM Clusters Backup Operation Status - Get"
+        ],
+        "operationId": "CloudHsmClustersBackupStatus_Get",
+        "description": "Gets the backup operation status of the specified Cloud HSM Cluster",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/cloudHsmClusterName"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/backupRestoreJobId"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Cloud HSM Cluster Backup Operation status returned",
+            "schema": {
+              "$ref": "#/definitions/BackupResult"
+            },
+            "headers": {
+              "x-ms-request-id": {
+                "description": "A unique ID for the current operation, service generated. All the resource providers must return this value in the response headers to facilitate debugging.",
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "The error response describing why the operation failed",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v6/types.json#/definitions/ErrorResponse"
+            },
+            "headers": {
+              "x-ms-request-id": {
+                "description": "A unique ID for the current operation, service generated. All the resource providers must return this value in the response headers to facilitate debugging.",
+                "type": "string"
+              }
+            }
+          }
+        },
+        "x-ms-examples": {
+          "CloudHsmCluster_Get_Backup_Status_MaximumSet_Gen": {
+            "$ref": "./examples/CloudHsmCluster_Backup_Pending_MaximumSet_Gen.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HardwareSecurityModules/cloudHsmClusters/{cloudHsmClusterName}/prerestore": {
```

Comment:
>prerestore [](http://example.com/codeflow?start=155&length=10)

Use an action name (verb or verb phrase). See earlier comment for prebackup action.

Reply:
We are following Azure Backup API convention, they are using PreBackup/PreRestore. https://msazure.visualstudio.com/One/_git/Mgmt-RecoverySvcs-WkloadCoord?path=/src/Documents/Swaggers/ArmBasedSwagger/ArmSwagger.json

Reply:
updated to use verb